.circleci/ci/src/jobs/job-setup.ts

```diff
@@ -0,0 +1,34 @@
+import { commands, Config, Job, reusable } from '@circleci/circleci-config-sdk';
+import { OpenJdkExecutor } from '../executors';
+import { Command } from '@circleci/circleci-config-sdk/dist/src/lib/Components/Commands/exports/Command';
+import { ReusedExecutor } from '@circleci/circleci-config-sdk/dist/src/lib/Components/Executors/exports/ReusedExecutor';
+import { orbs } from '../orbs';
+import { config } from '../config';
+import { CircleCIEnvironment } from '../pipelines';
+import { RestoreMavenCache } from '../commands';
+
+export class SetupJob {
+  public static create(dynamicConfig: Config, environment: CircleCIEnvironment): Job {
+    const openJdkExecutor = OpenJdkExecutor.get('small');
+    dynamicConfig.addReusableExecutor(openJdkExecutor);
+
+    const restoreMavenCacheCmd = RestoreMavenCache.get('job-build', environment);
+    dynamicConfig.addReusableCommand(restoreMavenCacheCmd);
+
+    const steps: Command[] = [
+      new commands.Checkout(),
+      new reusable.ReusedCommand(orbs.keeper.commands['env-export'], {
+        'secret-url': config.secrets.mavenSettings,
+        'var-name': 'MAVEN_SETTINGS',
+      }),
+      new commands.Run({
+        command: `echo $MAVEN_SETTINGS > ${config.maven.settingsFile} `,
+      }),
+      new commands.workspace.Persist({
+        root: '.',
+        paths: [config.maven.settingsFile],
+      }),
+    ];
+    return new Job('job-build-backend', new ReusedExecutor(openJdkExecutor), steps);
```

Comment:
The job name is not the right one.
```suggestion
    return new Job('job-setup', new ReusedExecutor(openJdkExecutor), steps);
```

On top of that, we don't need a JDK executor, as we don't use maven to determine the version. So probably the simplest executor is enough.